pkg/operations/kubernetesupgrade/upgrader.go

```diff
@@ -383,6 +385,43 @@ func (ku *Upgrader) upgradeAgentPools(ctx context.Context) error {
 }
 
 func (ku *Upgrader) upgradeAgentScaleSets(ctx context.Context) error {
+	if len(ku.ClusterTopology.AgentPoolScaleSetsToUpgrade) > 0 {
+		// need to apply the ARM template with target Kubernetes version to the VMSS first in order that the new VMSS instances
+		// created can get the expected Kubernetes version. Otherwise the new instances created still have old Kubernetes version
+		// if the topology dosn't have master nodes(so there is no ARM deployments in previous upgradeMasterNodes step)
```

Comment:
the last line should be `if the topology doesn't have master nodes (so there are no ARM deployments in previous upgradeMasterNodes step) `

Reply:
good catch, thank you. Changed in commit 4.